services/paymaster/src/db/config.rs

```diff
@@ -0,0 +1,177 @@
+use crate::config_manager::config::{Config, Domain};
+use crate::constraint::TransactionVariation;
+use crate::db::pool::pool;
+use std::collections::HashMap;
+use url::Url;
+
+use argon2::{
+    password_hash::{rand_core::OsRng, PasswordHasher, SaltString},
+    Argon2,
+};
+use sqlx::{types::Json, FromRow};
+use uuid::Uuid;
+
+use crate::db::pool;
+
+/// A row from the database that represents a domain and its transaction variations.
+#[derive(FromRow)]
+struct Row {
+    domain_id: Uuid,
+    domain: String,
+    enable_session_management: bool,
+    enable_preflight_simulation: bool,
+    instructions: Json<TransactionVariation>,
+}
+
+/// Load the config from the database
+pub async fn load_config() -> Result<Config, sqlx::Error> {
+    let rows: Vec<Row> = sqlx::query_as::<_, Row>(
+        r#"
+        SELECT
+          d.id            AS domain_id,
+          d.domain        AS domain,
+          d.enable_session_management,
+          d.enable_preflight_simulation,
+          v.instructions  AS instructions
+        FROM domain_config d
+        JOIN variation v ON v.domain_config_id = d.id
+        ORDER BY d.id
+        "#,
+    )
+    .fetch_all(pool::pool())
+    .await?;
+
+    let mut map: HashMap<Uuid, Domain> = HashMap::new();
```

Comment:
instead of making a mut map and adding to it via for loop, try a more functional approach a la:

```
let map: HashMap<Uuid, Domain> = rows.into_iter().fold(HashMap::new(), |mut acc, r| {
    acc.entry(r.domain_id)
        .or_insert_with(|| Domain {
            domain: r.domain.clone(),
            enable_session_management: r.enable_session_management,
            enable_preflight_simulation: r.enable_preflight_simulation,
            tx_variations: Vec::new(),
        })
        .tx_variations
        .push(r.instructions.0);
    acc
});
```